enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,458 @@
+course, even after merging, if any regression is found, we can easily revert the
+PR to get back to green.
+
+Regarding end-user code, although bootimages are by their very nature
+transient (we boot, and shortly after pivot into the node image), there are
+some situations where user-provided code may run (e.g. when manually extending
+`bootstrap.ign` or `master.ign` or when scripting the live ISO). This code may
+be relying on packages that are going away (e.g. some local-only `oc` command,
+or grepping `--help` texts of the kubelet, etc...).
+
+We shouldn't worry *too much* about breaking these use cases, but it still seems
+worth highlighting in the GA release notes that these OCP packages are no longer
+present.
+
+### Drawbacks
+
+- This will make it somewhat harder (or at the very least much more explicit) to
+  fast-track RHEL fixes into specific OCP releases. Because everything beneath
+  the node layer is RHEL-versioned and possibly shared across multiple OCP
+  releases, any fast-tracks would need to happen by doing an RPM upgrade within
+  the node layer. This is more cumbersome than just tagging packages into the
+  the ART plashets as currently done, but makes us more thorough in our known
+  deltas from RHEL.
+  - We should do this as a last resort and instead work more closely with RHEL
+    package maintainers to get fixes into the releases we need.
+
+## Open Questions [optional]
+
+- How to handle the early kernels we currently get from RHEL?
+  - The early kernels are RHEL-versioned and it would make sense to have them be
+    part of the CoreOS layer. (IOW, we always use the same early kernels for the set
+    of OCP streams sharing the same RHEL version.)
+  - Currently, this is done by tagging the kernel builds into a specific tag in
+    Brew and those builds then being added to the ART plashets. To avoid having
+    to source the (OCP-versioned) ART plashets during the CoreOS builds, we
+    could instead create new repos for these packages that are clearly defined
+    to be RHEL-versioned. The CoreOS build could then add that repo alongside
+    baseos and appstream.
+
+## Test Plan
+
+This has been covered in the proposal and implementation sections.
+
+This isn't an optional feature that's getting added; once we cut over to the
+new model, it will be implicitly continuously tested by all OpenShift CI tests.
+
+## Graduation Criteria
+
+This is not a new user-facing (or even internal-facing) OpenShift API. However,
+for completeness, let's consider "graduation" in this case as switching OCP over
+to using the layered image in production.
+
+For this to happen, the criteria are:
+- all affected OpenShift components and processes have been adapted to deal with
+  the new bootimages (which no longer have OpenShift packages)
+- the OpenShift installer now uses the new bootimages
+- the OpenShift node image can be built in Prow/Konflux
+- the new artifacts are used as part of the CI streams and all tests pass
+
```

Comment:
- Release controller updates for diff view. cc @bradmwilliams

Reply:
Thanks, added!